presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectBatchStreamReader.java

```diff
@@ -127,35 +141,40 @@ public Block readBlock()
         int[] offsetVector = new int[nextBatchSize + 1];
 
         if (presentStream == null) {
-            if (lengthStream == null) {
-                throw new OrcCorruptionException(streamDescriptor.getOrcDataSourceId(), "Value is not null but length stream is not present");
-            }
             lengthStream.nextIntVector(nextBatchSize, offsetVector, 0);
         }
         else {
             isNullVector = new boolean[nextBatchSize];
-            int nullValues = presentStream.getUnsetBits(nextBatchSize, isNullVector);
-            if (nullValues != nextBatchSize) {
+            int nullCount = presentStream.getUnsetBits(nextBatchSize, isNullVector);
+            if (nullCount == nextBatchSize) {
                 if (lengthStream == null) {
```

Comment:
This should no longer be needed since you've already handled this case in line 129-133?

Reply:
I think you're right. Removing this block.